src/stores/api/types/testutils/faker.go

```diff
@@ -94,18 +96,61 @@ func FakeSignTypedDataRequest() *types.SignTypedDataRequest {
 	}
 }
 
-func FakeSignETHTransactionRequest() *types.SignETHTransactionRequest {
+func FakeSignETHTransactionRequest(txType string) *types.SignETHTransactionRequest {
 	toAddress := common.HexToAddress("0x905B88EFf8Bda1543d4d6f4aA05afef143D27E18")
 
-	return &types.SignETHTransactionRequest{
-		Nonce:    0,
-		To:       &toAddress,
-		Value:    hexutil.Big(*hexutil.MustDecodeBig("0xfeee")),
-		GasPrice: hexutil.Big(*hexutil.MustDecodeBig("0xfeee")),
-		GasLimit: 21000,
-		ChainID:  hexutil.Big(*hexutil.MustDecodeBig("0x1")),
-		Data:     hexutil.MustDecode("0xfeee"),
+	switch txType {
+	case types.LegacyTxType:
+		return &types.SignETHTransactionRequest{
+			TransactionType: txType,
+			Nonce:           0,
+			To:              &toAddress,
+			Value:           hexutil.Big(*hexutil.MustDecodeBig("0xfeee")),
+			GasPrice:        hexutil.Big(*hexutil.MustDecodeBig("0xfeee")),
+			GasLimit:        21000,
+			Data:            hexutil.MustDecode("0xfeee"),
+			ChainID:         hexutil.Big(*hexutil.MustDecodeBig("0x1")),
+		}
+	case types.AccessListTxType:
+		return &types.SignETHTransactionRequest{
+			TransactionType: txType,
+			Nonce:           0,
+			To:              &toAddress,
+			Value:           hexutil.Big(*hexutil.MustDecodeBig("0xfeee")),
+			GasPrice:        hexutil.Big(*hexutil.MustDecodeBig("0xfeee")),
+			GasLimit:        21000,
+			ChainID:         hexutil.Big(*hexutil.MustDecodeBig("0x1")),
+			Data:            hexutil.MustDecode("0xfeee"),
+			AccessList: []ethtypes.AccessTuple{
+				{
+					Address:     toAddress,
+					StorageKeys: []common.Hash{common.HexToHash("0xfeee")},
+				},
+			},
+		}
+	case "", types.DynamicFeeTxType:
+		baseFee := hexutil.Big(*hexutil.MustDecodeBig("0xfeee"))
+		minerTip := hexutil.Big(*hexutil.MustDecodeBig("0xfeee"))
+		return &types.SignETHTransactionRequest{
+			TransactionType: txType,
+			Nonce:           0,
+			To:              &toAddress,
+			Value:           hexutil.Big(*hexutil.MustDecodeBig("0xfeee")),
+			GasFeeCap:       &baseFee,
+			GasTipCap:       &minerTip,
+			GasLimit:        21000,
+			ChainID:         hexutil.Big(*hexutil.MustDecodeBig("0x1")),
+			Data:            hexutil.MustDecode("0xfeee"),
+			AccessList: []ethtypes.AccessTuple{
+				{
+					Address:     toAddress,
+					StorageKeys: []common.Hash{common.HexToHash("0xfeee")},
+				},
+			},
+		}
 	}
```

Comment:
minor: It's hard to see in this case which is the difference between the returns.
Can we just have what's different in the cases, and after the switch the return with every parameters set correctly.
And a default case to `return nil` if no match is found?

Reply:
true! I mixed this faker with the formatter and didn't realise the type is the same in all 3 switches. Done.